### DISCUSSION TITLE
Simple emulation guide

### DISCUSSION THREAD

**Discussion by orel-binadev** [polhenarejos/pico-hsm#50] [Q&A] [answered]
Hi,
I'm a beginner in the embedded world, I'm trying to get started with emulation mode and would like to debug using vscode.
Could anyone point me to some sample projects or documentation that could help me set up and work with emulation mode? Any guidance or resources would be greatly appreciated.

Thanks in advance!

**Comment by polhenarejos** [comment]
Emulation mode produces a normal Linux/mac app, which can be debugged as a normal app with `gdb` or `valgrind`. Nothing special is required.

**Reply by orel-binadev** [reply]
Do you have a sample code on how to send simple apdu command?

**Reply by polhenarejos** [reply]
Pico HSM is not a software to communicate with an HSM, it is an HSM itself. It does not send APDU, it receives APDUs.

If you want to send APDU, use `OpenSC` for instance.

**Reply by dev-area** [reply]
so the HSM software runs as normal linux app but how to communicate with it? after build and run i can see (using strace) that its trying to connect to a TCP server , is there any server tool?

**Reply by polhenarejos** [reply, marked as answer]
You need vsmartcard-vpcd whicy can be installed through apt-get.

**Reply by dev-area** [reply]
thanks
